### PR TITLE
Create a useStaticValue hook

### PR DESCRIPTION
**Pull request by ThaNarie**

### CODE REVIEW

**Comment by mmnathan** [comment]
Did you consider a more descriptive name like `useInitialValue/useStaticValue/useIdentity` (ok the last one is too obscure)?

**Comment by mmnathan** [comment]
Should it also accept a direct (non-lazy) value to be consistent with `useState`? I can imagine this being useful when you'd like the initial value of a prop.

**Comment by ThaNarie** [comment]
> Did you consider a more descriptive name like useInitialValue/useStaticValue/useIdentity (ok the last one is too obscure)?

I didn't, but I like `useStaticValue` :)

> Should it also accept a direct (non-lazy) value to be consistent with useState? I can imagine this being useful when you'd like the initial value of a prop.

I guess that could be useful for that specific scenario, but you can already `useRef` for that. The main reason for this hook is to work with values that should be computed once (that's where useRef fails), or would create a new reference every time (that's where useMemo is cumbersome, and not 100% safe).

File: src/hooks/useValue/useValue.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { useRef } from 'react';
+
+/**
+ * Keep track of a value that is initialized once and then never changes.
+ *
+ * This behavior is similar to `useState`, but the value is never updated.
+ */
+export function useValue<T>(initializeFunction: () => T): T {
+  const valueRef = useRef<T | null>(null);
+  if (!valueRef.current) {
```
Comment (leroykorterink) [inline src/hooks/useValue/useValue.ts:10]: So if the initial value is falsy the value can still change? I think that shouldn't happen right?

I'd use the useState hook in the implementation.

```ts
const [value] = useState(initalizeFunction):
return value;
```

**Review by leroykorterink — CHANGES_REQUESTED** [review]

**Comment by ThaNarie** [comment]
@mmnathan I updated the implementation (and the name), and now it does support the same **initial value** as `useState` does :) More by accident because internally it uses `useState`.

**Review by leroykorterink — APPROVED** [review]

**Pull request merged**